ludwig/trainers/trainer.py

```diff
@@ -342,7 +347,8 @@ def train_for_tuning(
                 for output_feature_name, output_feature in self.model.output_features.items()
             }
             self.train_step(inputs, targets)
-        return self.model
+        duration_s = time.time() - start_ts
+        return batch_size / duration_s / total_steps
```

Comment:
isn't this supposed to be `batch_size * total_steps / duration_s`?

For example, if you have `batch_size=30` and `total_steps=30` and `duration_s=900`, wouldn't you want to say `total_samples=batch_size*total_steps=30*30=900`, making `samples_per_sec=total_samples/duration_s=900/900`, which is 1 sample per second? Could be missing something, but that was my impression of samples per second.

Reply:
The way I looked at it was

- `average_duration_per_step` = `duration_s` / `total_steps`
- `time_per_one_training_example` = `average_duration_per_step` / `batch_size`
- `throughput` = 1 / `time_per_one_training_example` = `batch_size` / `duration_s` / `total_steps`

Reply:
Yes, I think you're right. I forgot about order of operations. My original version was meant to be `batch_size / (duration_s / total_steps)`, which would be equivalent to your form.